### PR TITLE
Daniel/socket refactor

### PR DESCRIPTION
**Pull request by dantheli**
Refactored socket behavior and removed some routes that were no longer necessary

- use rooms for admins and users instead of storing clients in arrays
- use namespaces for different sessions instead of different ports
- ends poll when admin leaves
- actually disconnect sockets and remove them from the array when poll ends
- send question when user connects and a question exists, thus removed get question router
- attach poll manager to express app such that the app runs on a single port

### CODE REVIEW

File: src/routers/EndPollRouter.js [hunk 1]
```diff
@@ -23,11 +22,12 @@ class EndPollRouter extends AppDevRouter<Object> {
...
     }
 
-    await PollManager.endPoll(poll);
+    await req.app.pollManager.endPoll(poll);
     if (save === false) {
       await PollsRepo.deletePollById(id);
     }
-    return {};
+
+    return poll;
```
Comment (meganle) [inline src/routers/EndPollRouter.js]: Is it necessary to return the poll?

**Comment by meganle** [comment]
Everything looks good to me otherwise!

File: src/routers/EndPollRouter.js (hunk 1 above)
Comment (meganle) [reply src/routers/EndPollRouter.js]: Because it might be deleted on the server if the admin chooses not to save

Comment (dantheli) [reply src/routers/EndPollRouter.js]: Good point! I just didn't like an empty object as a response but that should be okay

**Pull request merged**